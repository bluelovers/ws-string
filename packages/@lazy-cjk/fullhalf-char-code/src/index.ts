
export const enum EnumFullHalfTableType
{
	FULL_WIDTH = 1,
	HALF_WIDTH = 0,
	NO_EXIST = -1,
}

/**
 * Determines whether the specified character code is full-width or half-width.
 */
export function detectFullHalfCharCode(charCode: number)
{
	if (0x0020 <= charCode && charCode < 0x007F)
	{
		return EnumFullHalfTableType.HALF_WIDTH;
	}

	if (0x3000 === charCode || 0xFF00 < charCode && charCode < 0xFF5F)
	{
		return EnumFullHalfTableType.FULL_WIDTH;
	}

	return EnumFullHalfTableType.NO_EXIST;
}

/**
 * Determines whether the specified character code is full-width or half-width.
 */
export function isFullHalfCharCode(charCode: number)
{
	const r = detectFullHalfCharCode(charCode);

	if (r === EnumFullHalfTableType.FULL_WIDTH)
	{
		return true;
	}
	else if (r === EnumFullHalfTableType.HALF_WIDTH)
	{
		return false;
	}
	else
	{
		// @todo add more...
	}

	return null;
}

/**
 * Converts a half-width character code to a full-width character code.
 * @param {number} charCode - The half-width character code to convert.
 * @returns {number} The full-width character code.
 */
export function toFullWidthCharCode(charCode: number)
{
	if (0x0020 < charCode && charCode < 0x007F)
	{
		return 0xFF00 + (charCode - 0x0020);
	}

	if (0x0020 === charCode)
	{
		return 0x3000;
	}

	return charCode;
}

/**
 * Converts a full-width character code to a half-width character code.
 * @param {number} charCode - The full-width character code to convert.
 * @returns {number} The half-width character code.
 */
export function toHalfWidthCharCode(charCode: number)
{
	if (0xFF00 < charCode && charCode < 0xFF5F)
	{
		return 0x0020 + (charCode - 0xFF00);
	}

	if (0x3000 === charCode)
	{
		return 0x0020;
	}

	return charCode;
}
