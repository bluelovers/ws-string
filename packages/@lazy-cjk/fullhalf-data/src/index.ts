import { tableFullHalfDefaultInclude } from './defaults';
import { tableFullHalf } from './table';

export interface ITableFullHalfObject
{
	from?: number;
	to?: number;

	values?: number[];

	not?: ITableFullHalfObject[],
}

export interface ITableFullHalf
{
	default?: ITableFullHalfObject;

	number?: ITableFullHalfObject;
	space?: ITableFullHalfObject;

	'A-Z'?: ITableFullHalfObject;
	'a-z'?: ITableFullHalfObject;

	not_default?: ITableFullHalfObject;

	[index: string]: ITableFullHalfObject;
}

export type ITableFullHalfDefaultIncludeKey = typeof tableFullHalfDefaultInclude[number]

export { tableFullHalf, tableFullHalfDefaultInclude }

export default tableFullHalf
