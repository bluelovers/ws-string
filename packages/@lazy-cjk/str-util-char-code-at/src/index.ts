/**
 * Created by user on 2017/12/9/009.
 */

export interface IcharCodeAtFn
{
	(char, charCode, str)
}

export function split(str): string[]
{
	return str.toString().split('');
}

export function charCodeAt(str, cb?: IcharCodeAtFn): number[]
{
	let ret = [];

	if (typeof cb !== 'function')
	{
		cb = void(0);
	}

	let _str = Array.isArray(str) ? str : str.toString();

	for (let char of _str)
	{
		let charCode = char.charCodeAt();
		let r;

		if (cb && (r = cb(char, charCode, str), typeof r != 'undefined'))
		{
			if (!r)
			{
				continue;
			}
			else if (Array.isArray(r))
			{
				ret = ret.concat(r);

				continue;
			}
		}

		ret.push(charCode);
	}

	return ret;
}

export default exports as typeof import('./util');
