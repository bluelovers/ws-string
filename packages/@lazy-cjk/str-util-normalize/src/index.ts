
export const StripTable = [

	/[\u2000-\u200F]/g,
	/[\u2028-\u202F]/g,
	/[\u205F-\u206F]/g,

	// ZERO WIDTH NO-BREAK SPACE
	/\uFEFF/g,

];

export function normalize(input: string, options: IOptions = {}): string
{
	if (!options.allow_nbsp)
	{
		input = input.replace(/\xA0/g, ' ');
	}

	if (!options.allow_bom)
	{
		input = input.replace(/\uFEFF/g, '');
	}

	StripTable.forEach(function (r)
	{
		input = input.replace(r, '');
	});

	return input;
}

export type IOptions = {
	allow_nbsp?: boolean,
	allow_bom?: boolean,
}

export default normalize;
