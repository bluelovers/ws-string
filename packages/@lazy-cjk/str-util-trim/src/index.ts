/**
 * Created by user on 2018/1/7/007.
 */

export interface ITrimOptions
{
	trim?: string,
	multiline?: boolean,

	no_start?: boolean,
	no_end?: boolean,
}

export function trim(txt: unknown, character_mask?: string): string
export function trim(txt: unknown, character_mask?: ITrimOptions): string
export function trim(txt: unknown, character_mask?: string | ITrimOptions): string
{
	let ret = txt.toString();

	if (typeof character_mask == 'number' || typeof character_mask == 'string')
	{
		character_mask = {
			trim: character_mask.toString(),
		};
	}

	if (character_mask)
	{
		let flags = '';

		if (character_mask.multiline)
		{
			flags += 'm';
		}

		let rs = [];
		let r;

		if (typeof character_mask.trim == 'string')
		{
			r = character_mask.trim.replace(/(\W)/g, '\\$1');

			if (character_mask.multiline)
			{
				r += ' \\t\\uFEFF\\xA0';
			}
			else
			{
				r += '\\s\\uFEFF\\xA0';
			}
		}
		else
		{
			if (character_mask.multiline)
			{
				r = ' \\t\\uFEFF\\xA0';
			}
			else
			{
				r = '\\s\\uFEFF\\xA0';
			}
		}

		if (!character_mask.no_start)
		{
			rs.push(`^[${r}]+`);
		}

		if (!character_mask.no_end)
		{
			rs.push(`[${r}]+$`);
		}

		rs = rs.map(r => new RegExp(r, flags));

		//console.log(rs);

		for (let r of rs)
		{
			ret = ret.replace(r, '');
		}
	}
	else
	{
		ret = ret
			.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, '')
		;
	}

	return ret;
}

export default trim

//console.log(trim('.  123  ?', '.?'));
