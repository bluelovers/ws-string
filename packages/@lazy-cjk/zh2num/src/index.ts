import { transcribeNumber, predefineedTranscriptionConfigs, transcriptionConfigs } from '@lazy-cjk/japanese';
import { characters, chinese_parseInt } from 'chinese-parseint2';

export { transcriptionConfigs };

export { predefineedTranscriptionConfigs };

export interface IOptions
{
	/**
	 * @see https://github.com/hakatashi/japanese.js/blob/master/src/numbers.es6
	 */
	digits?: string[];
	/**
	 * @see https://github.com/hakatashi/japanese.js/blob/master/src/numbers.es6
	 */
	unitNames?: string[];

	/**
	 * replace all
	 */
	unsafe?: boolean;
	/**
	 * RegExp Flags
	 */
	flags?: string;

	chars?: string;

	truncateOne?: number | boolean;

	strict?: boolean;

	one?: boolean;

	string?: boolean;
}

export const defaultOptions = {
	one: true,
} as IOptions;

/**
 * 將漢字轉換成數字
 * Parse Chinese/Japanese numeric strings to integer
 *
 * @param str
 * @param {IOptions} options
 * @returns {string|number}
 *
 * @example zh2num('千百十七') == 1117
 */
export function zh2num(str: number, options?: IOptions): number
export function zh2num(str: unknown, options?: IOptions): string
export function zh2num(str: any, options: IOptions = {})
{
	options = Object.assign({}, defaultOptions, options);

	let sa: string[] = [];
	let do_def = true;

	let jtc = transcriptionConfigs;
	let jpc = predefineedTranscriptionConfigs;

	for (let id of [
		'digits',
		'unitNames',
	] as const)
	{
		if (Array.isArray(options[id]) && options[id].length)
		{
			for (let d of options[id])
			{
				// @ts-ignore
				if (jpc[id][d])
				{
					// @ts-ignore
					sa = sa.concat(Object.values(jpc[id][d]));
				}
			}
		}
	}

	if (typeof options.truncateOne === 'number')
	{
		for (let i in jtc.default.truncateOne)
		{
			// @ts-ignore
			if (i < options.truncateOne)
			{
				sa.push(jtc.default.truncateOne[i])
			}
		}
	}
	else if ((typeof options.truncateOne === 'undefined' || options.truncateOne === true) && sa.length)
	{
		sa = sa.concat(Object.values(jtc.default.truncateOne));
	}

	if (options.chars)
	{
		sa = sa.concat(Array.isArray(options.chars) ? options.chars : options.chars.split(''));
	}

	if (sa.length)
	{
		do_def = false;
	}

	sa.filter(function (value)
	{
		if (value.length === 1 && (value in characters))
		{
			return true;
		}

		return false;
	});

	if (do_def)
	{
		sa = Object.keys(characters);
	}

	options.flags ||= 'u';

	if (options.unsafe)
	{
		options.flags += 'g';
	}

	//console.log(options, sa);

	let rs = sa.join('');

	let s = str.toString();

	if (!(new RegExp('([^' + rs + '])')).test(s))
	{
		return _chinese_parseInt(s, options) as number;
	}
	else if (options.strict)
	{
		return Number.NaN;
	}

	let r = new RegExp('([' + rs + ']+)', options.flags);

	// @ts-ignore
	return s.replace(r, function (...m)
	{
		return _chinese_parseInt(m[1], options) as any;
	}).toString();
}

const reChineseParseIntOne = /([佰百])([一二三四五六七八九壹貳參肆伍陸柒捌玖])(?![零○〇一二三四五六七八九壹貳參肆伍陸柒捌玖拾十什])/;

export function _chinese_parseInt(str: string, options: IOptions & {
	string: true,
}): string
export function _chinese_parseInt(str: string, options?: IOptions & {
	string?: false,
}): number
export function _chinese_parseInt(str: string, options?: IOptions): number
export function _chinese_parseInt(str: string, options: IOptions = {}): number | string
{
	if (options.one)
	{
		str = str.replace(reChineseParseIntOne, '$1〇$2');
	}

	let ret = chinese_parseInt(str);

	if (options.string)
	{
		// @ts-ignore
		ret = ret.toString();
	}

	return ret;
}

/**
 * 將數字轉為漢字
 *
 * @param number
 * @param options
 * @returns {string}
 *
 * @example num2zh(1117) == '千百十七'
 */
export function num2zh(number: unknown, options: {
	strict?: boolean,
} = {}): string
{
	let d = parseInt(number as any);

	if (Number.isNaN(d) || (options?.strict && (typeof number !== 'number' || d !== number)))
	{
		throw new TypeError(`${number} is not valid allow number`);
	}

	return transcribeNumber(d, options);
}

export default {
	zh2num,
	num2zh,
}
