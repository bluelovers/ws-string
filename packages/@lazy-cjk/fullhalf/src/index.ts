
import { EnumFullHalfTableType, toFullWidthCharCode, toHalfWidthCharCode } from '@lazy-cjk/fullhalf-char-code';
import UString from 'uni-string';
import { all } from 'deepmerge';
import { tableFullHalf, ITableFullHalfDefaultIncludeKey, tableFullHalfDefaultInclude, ITableFullHalfObject } from '@lazy-cjk/fullhalf-data';

let typeOnly: IOptions = {
	only: {
		number: true,
	},
};

export const toFullNumber = factory<string>(toFullWidthCharCode, EnumFullHalfTableType.FULL_WIDTH, typeOnly);
export const toHalfNumber = factory<string>(toHalfWidthCharCode, EnumFullHalfTableType.HALF_WIDTH, typeOnly);

typeOnly = {
	only: {
		eng: true,
	},
};

export const toFullEnglish = factory<string>(toFullWidthCharCode, EnumFullHalfTableType.FULL_WIDTH, typeOnly);
export const toHalfEnglish = factory<string>(toHalfWidthCharCode, EnumFullHalfTableType.HALF_WIDTH, typeOnly);

typeOnly = {
	only: {
		default: true,
	},
};

export const toFullWidth = factory<string>(toFullWidthCharCode, EnumFullHalfTableType.FULL_WIDTH, typeOnly);
export const toHalfWidth = factory<string>(toHalfWidthCharCode, EnumFullHalfTableType.HALF_WIDTH, typeOnly);

export function _optionsType(data: IOptionsType)
{
	if (data)
	{
		if (typeof data.exists === 'boolean')
		{
			for (let key in tableFullHalf[0])
			{
				if (key.indexOf('default') == 0)
				{
					continue;
				}

				if (data[key] !== false)
				{
					data[key] = data.exists;
				}
			}

			delete data.exists;
		}
		else
		{
			if (typeof data.default === 'boolean')
			{
				for (let key of tableFullHalfDefaultInclude)
				{
					if (data[key] !== false)
					{
						data[key] = data.default;
					}
				}

				delete data.default;
			}

			if (typeof data.not_default2 === 'boolean')
			{
				data.both = data.space = data.not_default2;
				delete data.not_default2;
			}

			if (typeof data.both === 'boolean')
			{
				data.number = data.eng = data.both;
				delete data.both;
			}

			if (typeof data.eng === 'boolean')
			{
				data['a-z'] = data['A-Z'] = data.eng;
				delete data.eng;
			}
		}
	}

	return data;
}

export function _chkType(charCode: number, data: ITableFullHalfObject)
{
	if (data.from && data.to && data.from <= charCode && charCode <= data.to)
	{
		return true;
	}
	else if (data.values?.includes(charCode))
	{
		return true;
	}
}

export function chkType(charCode: number, key: ITableFullHalfDefaultIncludeKey, type: EnumFullHalfTableType)
{
	let data: ITableFullHalfObject = tableFullHalf[type][key];

	//console.log(charCode, data);

	if (Array.isArray(data.not) && data.not.length)
	{
		for (let d of data.not)
		{
			if (_chkType(charCode, d))
			{
				return false;
			}
		}
	}

	if (_chkType(charCode, data))
	{
		return true;
	}
}

//@ts-ignore
export function process<T, U = string>(str: unknown, charProcess: ICharProcessor, options: IOptions)
{
	let ret: number[] = [];

	options.skip = _optionsType(options.skip);
	options.only = _optionsType(options.only);

	//console.log(options);

	let _str = Array.isArray(str) ? str : new UString(str);

	for (let char of _str)
	{
		let _skip: boolean;

		// @ts-ignore
		//let charCode = typeof char == 'number' ? char : char.charCodeAt();
		let charCode = typeof char == 'number' ? char : char.codePointAt();

		if (options.only)
		{
			_skip = true;

			for (let key in options.only)
			{
				//console.log(char, charCode, [key], chkType(charCode, key, options.type));

				if (options.only[key] && chkType(charCode, key as ITableFullHalfDefaultIncludeKey, options.type))
				{
					_skip = false;
					break;
				}
			}

			//console.log(char, charCode, _skip);
		}

		if (!_skip && options.skip)
		{
			for (let key in options.skip)
			{
				if (options.skip[key] && chkType(charCode, key as ITableFullHalfDefaultIncludeKey, options.type))
				{
					_skip = true;
					break;
				}
			}
		}

		if (_skip)
		{
			ret.push(charCode);
			continue;
		}

		ret.push(charProcess(charCode));
	}

	if (options.returnType)
	{
		return ret as number[];
	}

	//return String.fromCharCode.apply(String, ret);
	return String.fromCodePoint.apply(String, ret);
}

export function factory<T = string>(charProcessor: ICharProcessor,
	type: number | EnumFullHalfTableType,
	overwriteOptions?: IOptions
): IFactoryFn
{
	//const deepmerge = require('deepmerge');

	// @ts-ignore
	return (str, options?: IOptions) =>
	{
		options = all([
			{
				//skip: {},
			}, options || {}, overwriteOptions || {}, {
				type: type,
			},
		]);

		//console.log(options);

		return process<T>(str, charProcessor, options);
	};
}

export function _filterTable(data: ITableFullHalfObject)
{
	let _a: number[] = [];

	if (data.from && data.to)
	{
		for (let i = data.from; i <= data.to; i++)
		{
			_a.push(i);
		}
	}

	if (data.values)
	{
		_a = _a.concat(data.values)
	}

	if (Array.isArray(data.not) && data.not.length)
	{
		_a = _a.filter(function (charCode)
		{
			for (let d of data.not)
			{
				if (_chkType(charCode, d))
				{
					return false;
				}
			}

			return true;
		});
	}

	return _a;
}

export interface IOptionsType extends Partial<Record<ITableFullHalfDefaultIncludeKey, boolean>>
{
	eng?: boolean,
	number?: boolean,

	/**
	 * eng & number
	 */
	both?: boolean;

	space?: boolean;
	exists?: boolean;

	default?: boolean;
	not_default?: boolean;
	not_default2?: boolean;

	slash?: boolean;
	bracket?: boolean;

	[index: string]: boolean;
}

export interface IOptionsBase
{
	type?: number;

	skip?: IOptionsType;
	only?: IOptionsType;

	mode?: {
		only?: boolean,

		[index: string]: boolean;
	};
}

export type IOptionsTrue = IOptionsBase & {

	/**
	 * 回傳直接回傳陣列而不組合成字串
	 */
	returnType: ILazyTrue;

}

export type IOptionsFalse = IOptionsBase & {

	/**
	 * 回傳直接回傳陣列而不組合成字串
	 */
	returnType?: ILazyFalse;

}

export interface IOptions extends IOptionsBase
{

	/**
	 * 回傳直接回傳陣列而不組合成字串
	 */
	returnType?: ILazyTrue | ILazyFalse;

}

type ILazyTrue = true | 1;
type ILazyFalse = 0 | false | void | undefined | null;

export interface IFactoryFn
{
	(str: string | string[], options?: IOptionsFalse): string

	(str: (string | number)[], options: IOptionsTrue): number[]

	(str: unknown, options: IOptionsTrue): number[]

	(str: (string | number)[]): string

	(str: (string | number)[], options?: IOptionsFalse): string

	(str: unknown, options: IOptionsFalse): string

	(str: unknown, options: IOptions): string | number[]
}

export type ICharProcessor = (charCode: number) => number

export default {
	toFullNumber,
	toHalfNumber,

	toFullEnglish,
	toHalfEnglish,

	toFullWidth,
	toHalfWidth,
}
