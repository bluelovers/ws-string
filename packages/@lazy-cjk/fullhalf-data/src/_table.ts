import { charCodeAt } from '@lazy-cjk/str-util-char-code-at';

/**
 * @see https://en.wikipedia.org/wiki/Code_page_437
 */
export let _table = {
	default: {
		from: 0x0020 + 1,
		to: 0x007F - 1,
		values: [0x0020],
	},

	number: [0x0030, 0x0039],

	'A-Z': [0x0041, 0x005A],
	'a-z': [0x0061, 0x007A],

	space: [0x0020],

	slash: {
		values: charCodeAt(`/\\`),
	},

	bracket: {
		values: charCodeAt(`()[]{}`),
	},
} as const;
