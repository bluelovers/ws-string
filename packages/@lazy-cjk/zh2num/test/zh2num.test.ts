/**
 * Created by user on 2017/12/11/011.
 */

import { inspect } from 'util';
import { IOptions, zh2num } from '../src';

let tests = [
	[
		[
			'千百十七話百十七話百十七話\n百十七話百十七話百十七話',
			{
				digits: [
					'common',

				],

				truncateOne: false,
			},
		],
		'千百十7話百十七話百十七話\n百十七話百十七話百十七話',
	],
	[
		[
			'千百十七話百十七話百十七話\n百十七話百十七話百十七話',
			{
				digits: [
					'common',

				],

				strict: true,

				truncateOne: false,
			},
		],
		Number.NaN,
	],
	[
		[
			'千百十七',
			{
				digits: [
					'common',

				],

				strict: true,

				truncateOne: false,
			},
		],
		Number.NaN,
	],
	[
		[
			'千百十七話百十七話百十七話\n百十七話百十七話百十七話',
			{
				digits: [
					'common',
				],

				truncateOne: 1,
			},
		],
		'千百17話百十七話百十七話\n百十七話百十七話百十七話',
	],
	[
		[
			'千百十七話百十七話百十七話\n百十七話百十七話百十七話',
			{
				digits: [
					'common',
				],

				truncateOne: 2,
			},
		],
		'千117話百十七話百十七話\n百十七話百十七話百十七話',
	],
	[
		[
			'千百十七話百十七話百十七話\n百十七話百十七話百十七話',
		],
		'1117話百十七話百十七話\n百十七話百十七話百十七話',
	],
	[
		[
			'千百十七',
			{
				digits: [
					'common',
				],
				truncateOne: 2,
				strict: true,
			},
		],
		Number.NaN,
	],
	[
		[
			'千百十七',
			{
				strict: true,
			},
		],
		1117,
	],
	[
		[
			'百二話',
			{
				one: false,
			},
		],
		'120話',
	],
	[
		[
			'百二話',
			{
				one: true,
			},
		],
		'102話',
	],
	[
		[
			'百二話',
			{},
		],
		'102話',
	],
];

describe(`zh2num`, () =>
{
	const fn = zh2num;

	tests.forEach(function (test, index)
	{
		it(`${test[0][0]} ${inspect(test[0][1])} => ${test[1]}`, () =>
		{
			//console.log('it:inner', currentTest.title);
			//console.log('it:inner', currentTest.fullTitle());

			// @ts-ignore
			let v = fn.apply(null, test[0]);

			expect(v).toEqual(test[1]);

			expect(typeof v).toMatchSnapshot();
			expect(v).toMatchSnapshot();

		});
	});
});

