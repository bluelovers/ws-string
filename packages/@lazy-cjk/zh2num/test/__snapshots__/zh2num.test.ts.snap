// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zh2num 千百十七 { digits: [ 'common' ], strict: true, truncateOne: false } => NaN 1`] = `"number"`;

exports[`zh2num 千百十七 { digits: [ 'common' ], strict: true, truncateOne: false } => NaN 2`] = `NaN`;

exports[`zh2num 千百十七 { digits: [ 'common' ], truncateOne: 2, strict: true } => NaN 1`] = `"number"`;

exports[`zh2num 千百十七 { digits: [ 'common' ], truncateOne: 2, strict: true } => NaN 2`] = `NaN`;

exports[`zh2num 千百十七 { strict: true } => 1117 1`] = `"number"`;

exports[`zh2num 千百十七 { strict: true } => 1117 2`] = `1117`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 { digits: [ 'common' ], strict: true, truncateOne: false } => NaN 1`] = `"number"`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 { digits: [ 'common' ], strict: true, truncateOne: false } => NaN 2`] = `NaN`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 { digits: [ 'common' ], truncateOne: 1 } => 千百17話百十七話百十七話
百十七話百十七話百十七話 1`] = `"string"`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 { digits: [ 'common' ], truncateOne: 1 } => 千百17話百十七話百十七話
百十七話百十七話百十七話 2`] = `
"千百17話百十七話百十七話
百十七話百十七話百十七話"
`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 { digits: [ 'common' ], truncateOne: 2 } => 千117話百十七話百十七話
百十七話百十七話百十七話 1`] = `"string"`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 { digits: [ 'common' ], truncateOne: 2 } => 千117話百十七話百十七話
百十七話百十七話百十七話 2`] = `
"千117話百十七話百十七話
百十七話百十七話百十七話"
`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 { digits: [ 'common' ], truncateOne: false } => 千百十7話百十七話百十七話
百十七話百十七話百十七話 1`] = `"string"`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 { digits: [ 'common' ], truncateOne: false } => 千百十7話百十七話百十七話
百十七話百十七話百十七話 2`] = `
"千百十7話百十七話百十七話
百十七話百十七話百十七話"
`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 undefined => 1117話百十七話百十七話
百十七話百十七話百十七話 1`] = `"string"`;

exports[`zh2num 千百十七話百十七話百十七話
百十七話百十七話百十七話 undefined => 1117話百十七話百十七話
百十七話百十七話百十七話 2`] = `
"1117話百十七話百十七話
百十七話百十七話百十七話"
`;

exports[`zh2num 百二話 { one: false } => 120話 1`] = `"string"`;

exports[`zh2num 百二話 { one: false } => 120話 2`] = `"120話"`;

exports[`zh2num 百二話 { one: true } => 102話 1`] = `"string"`;

exports[`zh2num 百二話 { one: true } => 102話 2`] = `"102話"`;

exports[`zh2num 百二話 {} => 102話 1`] = `"string"`;

exports[`zh2num 百二話 {} => 102話 2`] = `"102話"`;
